pkg/apis/networking/validation/validation.go

```diff
@@ -385,33 +396,64 @@ func validateHTTPIngressPath(path *networking.HTTPIngressPath, fldPath *field.Pa
 		}
 	}
 
-	allErrs = append(allErrs, validateIngressBackend(&path.Backend, fldPath.Child("backend"), opts)...)
+	allErrs = append(allErrs, validateIngressBackend(&path.Backend, fldPath.Child("backend"), opts, requestGV)...)
 	return allErrs
 }
 
+// portFieledName returns the name of the field used for an
+// ingress service port in the provided GroupVersion.
+func portFieldName(gv schema.GroupVersion) string {
+	switch gv {
+	case networkingv1beta1.SchemeGroupVersion, extensionsv1beta1.SchemeGroupVersion:
+		return "servicePort"
+	default:
+		return "port"
+	}
+}
+
+// serviceName returns the name of the field used for a
+// service in the provided GroupVersion.
+func serviceName(gv schema.GroupVersion) string {
```

Comment:
probably take the backend field path as input, and return the field path as output, since different versions have different nesting levels:

```
func serviceNameFieldPath(backendFieldPath *field.Path, gv schema.GroupVersion) *field.Path {
  ...
  return backendFieldPath.Child("serviceName")
  ...
  return backendFieldPath.Child("service", "name")
}
```

similar for servicePortNameFIeldPath and servicePortNumberFieldPath